designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceImpl.scala

```diff
@@ -259,10 +260,15 @@ class DeploymentServiceImpl(dispatcher: DeploymentManagerDispatcher,
 
   //We assume that checking the state for archived doesn't make sense, and we compute the state based on the last state action
   private def getArchivedProcessState(processDetails: BaseProcessDetails[_])(implicit manager: DeploymentManager) = {
-    processDetails.lastStateAction match {
-      case Some(_) => //We assume that cancellation is required before the archiving process
+    processDetails.lastStateAction.map(_.action) match {
+      case Some(Cancel) =>
         logger.debug(s"Status for: '${processDetails.name}' is: ${SimpleStateStatus.Canceled}")
         DBIOAction.successful(manager.processStateDefinitionManager.processState(SimpleStateStatus.Canceled))
+      case Some(Deploy) =>
```

Comment:
IMO this Some(Deploy) and Some(_) is the same case. We have some inconsistency and we should decide how to handle it. One way is thow an exception and let's user inform admin to fix this. Other way is to return `Problem` and log error. Pick one of them.